src/xrc/src/forex.rs

```diff
@@ -1795,6 +1883,140 @@ mod test {
         );
     }
 
+    #[test]
+    fn rate_store_gets_rate_in_past_if_current_day_is_not_over() {
+        // Create a store, update, check that only rates with more sources were updated.
+        let mut store = ForexRateStore::new();
+        // Rate 0
+        store.put(
+            0,
+            hashmap! {
+                "EUR".to_string() =>
+                    QueriedExchangeRate {
+                        base_asset: Asset {
+                            symbol: "EUR".to_string(),
+                            class: AssetClass::FiatCurrency,
+                        },
+                        quote_asset: Asset {
+                            symbol: USD.to_string(),
+                            class: AssetClass::FiatCurrency,
+                        },
+                        timestamp: 0,
+                        rates: vec![800_000_000],
+                        base_asset_num_queried_sources: 4,
+                        base_asset_num_received_rates: 4,
+                        quote_asset_num_queried_sources: 4,
+                        quote_asset_num_received_rates: 4,
+                        forex_timestamp: Some(1234),
```

Comment:
This is not a problem here but we don't want forex timestamps in the future.

Reply:
Fixed the timestamps to be the same as the day.